Fix clippy issues in tests

kurtjd opened the pull request:
Running `cargo clippy --tests` revealed several issues. This PR addresses those issues.

Of note is that the `style` lint was changed to `deny`. This is because clippy internally overrides a false positive of `needless_return` which conflicts with `forbid`. The generated error message notes this will soon become a hard error and links to [this issue](https://github.com/rust-lang/rust/issues/81670) recommending replacing `forbid` with `deny`.

jerrysxie commented:
> Running `cargo clippy --tests` revealed several issues. This PR addresses those issues.
> 
> Of note is that the `style` lint was changed to `deny`. This is because clippy internally overrides a false positive of `needless_return` which conflicts with `forbid`. The generated error message notes this will soon become a hard error and links to [this issue](https://github.com/rust-lang/rust/issues/81670) recommending replacing `forbid` with `deny`.

Should we consider adding a workflow to run `cargo clippy --test` as well to make sure that this does not reoccur in the future?

kurtjd commented:
> > Running `cargo clippy --tests` revealed several issues. This PR addresses those issues.
> > Of note is that the `style` lint was changed to `deny`. This is because clippy internally overrides a false positive of `needless_return` which conflicts with `forbid`. The generated error message notes this will soon become a hard error and links to [this issue](https://github.com/rust-lang/rust/issues/81670) recommending replacing `forbid` with `deny`.
> 
> Should we consider adding a workflow to run `cargo clippy --test` as well to make sure that this does not reoccur in the future?

I think that's a good idea. I'll see if I can add it to this commit.

kurtjd commented:
> > Running `cargo clippy --tests` revealed several issues. This PR addresses those issues.
> > Of note is that the `style` lint was changed to `deny`. This is because clippy internally overrides a false positive of `needless_return` which conflicts with `forbid`. The generated error message notes this will soon become a hard error and links to [this issue](https://github.com/rust-lang/rust/issues/81670) recommending replacing `forbid` with `deny`.
> 
> Should we consider adding a workflow to run `cargo clippy --test` as well to make sure that this does not reoccur in the future?

I've updated the workflow to have clippy check all targets (including test) and all features. I intentionally kept one of the arguments as `-F clippy::style` (instead of `-D clippy::style`) to ensure a clippy warning would be generated. We can see it did indeed get caught by the workflow: https://github.com/OpenDevicePartnership/tmp108/actions/runs/15766043203/job/44442594830?pr=20#step:4:24

However, for some reason it's not counting as a failed check and raising any workflow issues. The same thing is happening on other ODP repos, whereas it works as expected on my personal forks. Might have to investigate if there are some repo settings causing this.

jerrysxie commented:
@kurtjd it seems like this PR needs to be updated after https://github.com/OpenDevicePartnership/tmp108/pull/21 has been merged,
